app/src/main/java/com/jvoye/tasky/core/domain/AuthInfo.kt

```diff
@@ -0,0 +1,7 @@
+package com.jvoye.tasky.core.domain
+
+data class AuthInfo(
+    val accessToken: String? = null,
+    val refreshToken: String? = null,
+    val userId: String? = null
+)
```

Comment:
If you want to indicate a whole object doesn't exist, it makes more sense to model that whole object as nullable. Otherwise, you'll find yourself doing null-checks all day, even though you'd only need one. 

So this...
```
data class AuthInfo(
    val userId: String,
    val accessToken: String,
    val refreshToken: String
)

// ...
var user: AuthInfo? = null
```
makes more sense than this...
```
data class AuthInfo(
    val userId: String?,
    val accessToken: String?,
    val refreshToken: String?
)

// ...
var authInfo: AuthInfo = AuthInfo()
```
Here, you'd model that an info exists, even though it doesn't and it ends up with null fields.

For the DataStore serializable it's necessary to model it like you did AFAIK, but not for the domain model